src/rgw/rgw-restore-bucket-index

```diff
@@ -23,35 +23,48 @@
 trap "clean ; exit 1" TERM
 export TOP_PID=$$
 
+# insure if any command in a pipe sequence fails, the exist status is reflected in $?
+set -o pipefail
+
 # IMPORTANT: affects order produced by 'sort' and 'ceph-diff-sorted'
 # relies on this ordering
 export LC_ALL=C
 
-# temporary files
-export bkt_entry=/tmp/rgwrbi-bkt-entry.$$
-export bkt_inst=/tmp/rgwrbi-bkt-inst.$$
-export marker_ls=/tmp/rgwrbi-marker-ls.$$
-export obj_list=/tmp/rgwrbi-object-list.$$
-export obj_list_ver=/tmp/rgwrbi-object-list-ver.$$
-export obj_reindex_script=/tmp/rgwrbi-object-list-script.$$
-export zone_info=/tmp/rgwrbi-zone-info.$$
-export olh_info_enc=/tmp/rgwrbi-olh-info-enc.$$
-export olh_info_json=/tmp/rgwrbi-olh-info-json.$$
-export debug_log=/tmp/rgwrbi-debug-log.$$
-
+# whether or not the temporary files are cleaned on completion
 export clean_temps=1
 
 # make explicit tabs easier to see in code
 export TAB="	"
 
-
+# cleans all temporary files
 clean() {
   if [ "$clean_temps" == 1 ] ;then
     rm -f $bkt_entry $bkt_inst $marker_ls $obj_list $obj_list_ver \
        $obj_reindex_script $zone_info $olh_info_enc $olh_info_json
   fi
 }
 
+test_temp_space() {
+    # if there is a command-line argument, assume it's exit status
+    # use exit status if provided, otherwise ignore
+    test_space=0
+    if [ $# -eq 0 ] ;then
+	test_space=1
+    elif [ "$1" -ne 0 ] ;then
```

Comment:
I'm not sure why we'd want to call `test_temp_space()` with no intention of actually testing the space. :)

Reply:
My original intent was to only test space when the command-line that was writing output there exited with an error code (i.e., non-zero). But then I had to relax that once I realized that there were problems if the exit status was generated by an "eval" or possibly whe a more complex control structure was used (output comes from an entire while loop).

But you're right; let's not even test the exit status and just check disk space.

Thanks, @linuxkidd .